packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1207,6 +1259,34 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   }
 }
 
+/// A helper class to contain a semantics node and the effective
+/// sort order of that node during the traversal of the tree. This is
+/// what is actually sorted when semantics nodes are sorted.
+class _TraversalSortNode implements Comparable<_TraversalSortNode> {
+  _TraversalSortNode(this.node, this.order);
+  SemanticsNode node;
+  SemanticsSortOrder order;
+
+  int _compareGeometry(_TraversalSortNode other) {
+    // TODO(gspencer): Move the geometric comparison from the platform side to here.
```

Comment:
I was under the impression that the accompanying engine PR completely removed geometric comparison from the platform side? What's left to move?

Reply:
And without that, what is actually now breaking ties?

Reply:
As we discussed, nothing is breaking ties.  The order of a tie is currently arbitrary, until I've implemented the geometric comparison again on the framework side.  It doesn't help to continue to compare geometry on the engine side, since they're all assigned `nextNodeId`s already.